### ISSUE TITLE
Clang tidy warning modernize-use-std-print does not work on sample code from documentation

### ISSUE THREAD

**Issue by belobrov-andrey** [llvm/llvm-project#129494]
The following minimal code sample from [documentation](https://clang.llvm.org/extra/clang-tidy/checks/modernize/use-std-print.html)

```
void getDescription(const std::string &description, int value)
{
    fprintf(stderr, "The %s is %3d\n", description.c_str(), value);
}
```
There is warning: unable to use 'std::format' instead of 'StrFormat' because **first argument is not a narrow string literal** [modernize-use-std-format]

Environment: Clang tidy 19, OS X, compile flags: -m64 -stdlib=libc++ -g -std=gnu++2b -arch x86_64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk

**Comment by belobrov-andrey** [comment]
Related to issue  https://github.com/llvm/llvm-project/issues/129484
